src/impl/apis/data_api_impl.py

```diff
@@ -313,135 +448,66 @@ async def run_query(
+    metadata = await get_metadata(query_parameters, tr, pool, token_model)
+    query_stream_generator = stream_query(base_table_name, metadata, pool, table_name)
     if query_parameters.category_id.lower() in ["residents", "presence"]:
-        out = region_to_csv(result.data_by_date)
+        async for csv_chunk in stream_region_to_csv(query_stream_generator):
+            yield csv_chunk
     elif query_parameters.category_id.lower() in ["relocations", "movements"]:
-        out = flows_to_csv(result.data_by_date)
+        async for csv_chunk in stream_flows_to_csv(query_stream_generator):
+            yield csv_chunk
     else:
         raise HTTPException(
             status_code=HTTPStatus.NOT_IMPLEMENTED,
             detail=f"CSV not yet implemented for {query_parameters.category_id.lower()}",
         )
-    return out
-
-
-def region_to_csv(region_data: Dict[str, Dict[str, float]]) -> str:
-    csv_rows = []
-    for date, data in region_data.items():
-        for source_region, value in data.items():
-            row = {"date": date, "area_code": source_region, "value": value}
-            csv_rows.append(",".join(str(v) for v in row.values()))
-    csv_header = ",".join(row.keys())
-    csv_out = "\r\n".join((csv_header, *csv_rows))
-    return csv_out
-
-
-def flows_to_csv(flow_data: Dict[str, Dict[str, Dict[str, float]]]) -> str:
-    csv_rows = []
-    for date, data in flow_data.items():
-        for source_region, value in data.items():
-            for dest_region, flow_value in value.items():
-                row = {
-                    "date": date,
-                    "origin_code": source_region,
-                    "destination_code": dest_region,
-                    "value": flow_value,
-                }
-                csv_rows.append(",".join(str(v) for v in row.values()))
-    csv_header = ",".join(row.keys())
-    csv_out = "\r\n".join((csv_header, *csv_rows))
-    return csv_out
+
+
+async def stream_region_to_csv(
+    region_stream: AsyncGenerator,
+) -> AsyncGenerator[str, None]:
+    yield "date,area_code,value\r\n"
+    async for chunk in region_stream:
+        if chunk:
+            row_str = ""
+            for row in chunk:
+                date = row[-1].strftime("%Y-%m-%d")
+                area_code = row[2]
+                value = str(row[3])
+                row_str += ",".join(str(v) for v in [date, area_code, value]) + "\r\n"
+            yield row_str
+    yield "\r\n"
+
+
+async def stream_flows_to_csv(flow_stream: AsyncGenerator) -> AsyncGenerator[str, None]:
+    yield "date,origin_code,destination_code,value\r\n"
+    async for chunk in flow_stream:
+        if chunk:
+            row_str = ""
+            for row in chunk:
+                date = row[-1].strftime("%Y-%m-%d")
+                orig_code = row[2]
+                dest_code = row[3]
+                value = str(row[4])
+                row_str += ",".join(str(v) for v in [date, orig_code, dest_code, value]) + "\r\n"
```

Comment:
```suggestion
                row_str = f"{row[-1]:%Y-%m-%d},{row[2]},{row[3]},{row[4]}\r\n"
```

Is going to be faster.

Reply:
Or even just yield and skip the assign actually.

Reply:
Did it this way for readability - I don't like referring to items by their index in an arbitary list if I can avoid it (future refactoring should probably involve changing `SSCursor` to `SSDictCursor` to get the best of both worlds

Reply:
This yields chunks, not rows now

Reply:
Ordinarily I would agree, but we care a lot about speed in this loop.

Reply:
(Because every trip through is time the main thread is blocked)